src/simplewallet/simplewallet.cpp

```diff
@@ -8255,7 +8525,67 @@ void simple_wallet::wallet_idle_thread()
 
     if (!m_idle_run.load(std::memory_order_relaxed))
       break;
-    m_idle_cond.wait_for(lock, boost::chrono::seconds(90));
+
+    if (m_rpc_payment_mining_requested || m_wallet->auto_mine_for_rpc_payment_threshold() >= 0)
+    {
+      uint64_t target = m_wallet->credits_target();
+      if (target == 0)
+        target = CREDITS_TARGET;
+      if (m_rpc_payment_mining_requested)
+        target = std::numeric_limits<uint64_t>::max();
+      bool need_payment = m_need_payment || m_rpc_payment_mining_requested || (m_wallet->credits() < target && m_wallet->daemon_requires_payment());
+      if (need_payment)
+      {
+        const boost::posix_time::ptime start_time = boost::posix_time::microsec_clock::universal_time();
+        bool ret = m_wallet->search_for_rpc_payment(target, [this](uint64_t diff, uint64_t credits_per_hash_found) {
```

Comment:
Can this at least be broken up so that the two lambdas are created on the stack for readability? This is a giant function call.

Reply:
Heh, OK.